src/middlewared/middlewared/plugins/pool_/dataset_summary.py

```diff
@@ -0,0 +1,31 @@
+from middlewared.schema import accepts, returns, Dict, Int, Str
+from middlewared.service import Service
+
+
+class PoolDatasetService(Service):
+
+    class Config:
+        namespace = "pool.dataset"
+
+    @accepts(Str("dataset"))
+    @returns(Dict(
+        Int("cloud_sync_task_count"),
+        Int("replication_task_count"),
+        Int("rsync_task_count"),
+        Int("snapshot_count"),
+        Int("snapshot_task_count"),
+    ))
+    async def summary(self, dataset):
+        path = f"/mnt/{dataset}"
+
```

Comment:
You need to convert dataset to mountpoint here. If not mounted, then check for dataset existence before creating synthetic path. Otherwise, you'll give bogus results for non-existent datasets.